.github/workflows/my-ci-pipeline.yml

```diff
@@ -77,5 +80,18 @@ jobs:
       - name: Teraform initialize the working directory
         run: terraform init
 
-      - name: Apply teraform changes
-        run: terraform apply
+      - name: Set up environment variables
```

Comment:
You'll need to do this stage _before_ the TF init, since that also relies on having permission to speak to Azure.

That said, personally I prefer just specifying them in the env block, e.g. with:
```
    env:
      ARM_CLIENT_ID: ${{ secrets.ARM_CLIENT_ID }}
      ARM_CLIENT_SECRET=${{ secrets.ARM_CLIENT_SECRET }}
      ARM_TENANT_ID=${{ secrets.ARM_TENANT_ID }}
      ARM_SUBSCRIPTION_ID=${{ secrets.ARM_SUBSCRIPTION_ID }}
      TF_VAR_prefix: "cohort28"
```

(That said, are there not other variables we need to specify here? E.g. oauth IDs and secrets?)